Upgrade react native 0.79.4 + Removed Mixpanel from Native Android/IOS SDKs

dporwal-shipit opened the pull request.

aaditya-smallcase committed:
fix: setup sdk function working now

indrajit-roy-sc committed:
feat: remove old example app and add new smart investing app

Use react-native 0.79.4 to setup the app

indrajit-roy-sc committed:
chore: android working state with 0.79.4

Updated the version of react-native-screens to ^3.34.0 in package.json and modified the Android build.gradle to include jcenter in the repositories list. Also, adjusted gradle.properties to disable new architecture and added artifactory credentials.

indrajit-roy-sc committed:
chore: ios working state with 0.79.4

Swapped the API categories in PrivacyInfo.xcprivacy and updated the project.pbxproj to include PrivacyInfo.xcprivacy in the Resources build phase. Adjusted build settings to include new flags and development team information.

indrajit-roy-sc committed:
feat: implement core functionality for smart investing app

Added multiple screens and components for the smart investing app, including ConnectScreen, HoldingsScreen, and SmtScreen. Introduced context providers for environment and SST cart management. Integrated API functions for handling user interactions and data fetching. Updated package.json to include react-native-gesture-handler and other dependencies. Enhanced the app's structure with new models and services for better data management.

indrajit-roy-sc committed:
chore: revert back iOS min version and SCGateway dependency to match
current prod values

indrajit-roy-sc committed:
chore: update Podfile.lock

Add cdn url for cocoapods specs

dporwal-shipit committed:
feat(analytics): Adding events capturing through new RCTBridgeModule

dporwal-shipit committed:
feat(analytics): Adding events capturing through new RCTBridgeModule

dporwal-shipit committed:
feature(analytics): working state

indrajit-roy-sc committed:
chore: update SCGateway dependency to version 4.1.5 in Podfile.lock

indrajit-roy-sc committed:
chore: update iOS run script to list available devices

indrajit-roy-sc committed:
revert: change impl. back to stable version

We had changed parameter types for isLeprechaunActive, isAmoEnabled, and showLoginCta from BOOL* to BOOL in setConfigEnvironment and triggerLeadGenWithLoginCta methods for supporting the new architecture.

We are reverting these changes so that we can first thoroughly validate these changes against both the old/new ARCHS and then proceed.

dporwal-shipit committed:
feature(analytics): updated code to capture events.

dporwal-shipit committed:
feature(analytics): updated code to capture events.

dporwal-shipit committed:
feat(analytics):- remove Mixpanel from SCLoans and substitute with new tracking
- Replaced Mixpanel tracking calls in SCLoans with internal analytics events
- Removed Mixpanel SDK dependency from gradle
- Verified build and functionality on working version

dporwal-shipit committed:
app building

dporwal-shipit committed:
developing android event flow

dporwal-shipit committed:
feature(analytics): working state android and ios scgateway emitters

dporwal-shipit committed:
feature(analytics): Working state android scloans events incoming

dporwal-shipit committed:
feature(analytics): working state ios events emission-capturing

dporwal-shipit committed:
feature(analytics): working state android events emission-capturing

dporwal-shipit commented:
React-Native Bridge for Android successfully propagating events from Native Android SDK(Producer) to Smart Investing Appln (Consumer):

For SCGateway:
<img width="720" height="468" alt="image" src="https://github.com/user-attachments/assets/ea26bbe4-32e7-4114-b9c3-e24b2cc40dc1" />

For SCLoans:
<img width="720" height="468" alt="image" src="https://github.com/user-attachments/assets/2a558ea4-f0ee-445e-a710-416521b669b0" />





React-Native Bridge for IOS successfully propagating events from Native Android IOS(Producer) to Smart Investing Appln (Consumer):

For SCGateway:
<img width="720" height="468" alt="image" src="https://github.com/user-attachments/assets/ca57b3a1-c67a-4dc6-82d7-da05282a5d5b" />

For SCLoans:
<img width="720" height="468" alt="image" src="https://github.com/user-attachments/assets/eedc5dc0-86c2-4e84-8e75-e667df1d2dee" />

dporwal-shipit committed:
refactor(analytics): restructure event bridging logic for improved maintainability
- Simplified event flow between native and JS layers
- Improved code readability and reduced complexity
- Easier to extend with new analytics events in future

dporwal-shipit committed:
refactor(analytics): restructure loans event bridging logic for improved maintainability
- Simplified event flow between native and JS layers
- Improved code readability and reduced complexity
- Easier to extend with new analytics events in future

LonelyCpp commented on smart_investing_react_native/App.tsx:
install the "prettier" vscode extention for formatting js files. make sure that it matches the older style

(here the diff should only show changes that you've already made)

dporwal-shipit replied:
Error: Cannot find module 'prettier' from '/Users/dhruvporwal/smallcase/react-native-smallcase-gateway'
    at e.exports [as sync] (/Users/dhruvporwal/.vscode/extensions/esbenp.prettier-vscode-11.0.0/dist/extension.js:1:38498)
    
 Unable to do for .js files. Would i have to add prettier to project? Because i have added prettier as default formatter in settings.json already:
    
    {
  "security.workspace.trust.untrustedFiles": "open",
  "dart.flutterSdkPath": "/Users/dhruvporwal/fvm/versions/stable",
  "git.enableSmartCommit": true,
  "files.autoSave": "afterDelay",
  "diffEditor.ignoreTrimWhitespace": true,
  "gitlens.ai.model": "gitkraken",
  "gitlens.ai.gitkraken.model": "gemini:gemini-2.0-flash",
  "dart.debugExternalPackageLibraries": true,
  "dart.debugSdkLibraries": true,
  "editor.formatOnSave": true,
  "prettier.resolveGlobalModules": true,
  "editor.formatOnPaste": true,
  "editor.formatOnType": true,
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "notebook.defaultFormatter": "esbenp.prettier-vscode"
}

LonelyCpp commented on src/SmallcaseGateway.js:
```diff
@@ -259,25 +233,56 @@ const triggerLeadGenWithLoginCta = async (
 /**
  * Returns the native android/ios and react-native sdk version
- * (internal-tracking)
- * @returns {Promise}
```
why did we remove these return and param types?

dporwal-shipit replied:
They have been added back.

LonelyCpp commented on src/SmallcaseGateway.js:
```diff
@@ -259,25 +233,56 @@ const triggerLeadGenWithLoginCta = async (
+
+/**
+ * 🔕 Unsubscribe from Gateway Event
+ * @param {object} subscription - Subscription returned from startGatewayEventListening
```
these param and return types can be better. make it closer to the actual types, instead of generic types

dporwal-shipit replied:
Done.

LonelyCpp commented on src/SCLoansEventEmitter.js:
```diff
@@ -0,0 +1,115 @@
+        this.isInitialized = true;
+        console.log('[SCLoansEvents] Initialized for', Platform.OS);
+      } else {
+        console.warn('[SCLoansEvents] Native module not found for', Platform.OS);
```
this probably should be thrown as an error. shouldn't silently fail

dporwal-shipit replied:
Done

LonelyCpp commented on src/SCLoansEventEmitter.js:
```diff
@@ -0,0 +1,115 @@
+    }
+
+    try {
+      const subscription = this.eventEmitter.addListener('scloans_notification', (eventData) => {
```
keep all these strings ("scloans_notification") as constants. don't use them directly

dporwal-shipit replied:
Done

LonelyCpp commented on src/SCLoansEventEmitter.js:
```diff
@@ -0,0 +1,115 @@
+        }
+
+        const normalizedEvent = {
+          type: eventData.eventType || eventData.type || 'unknown_event',
+          eventType: eventData.eventType || eventData.type || 'unknown_event',
+          data: eventData.data || eventData,
+          timestamp: eventData.timestamp || Date.now(),
```
do these fallbacks make sense? 

please check with the analytics team on what should be the correct fallback values such that invalid values don't make it into real analysis

dporwal-shipit replied:
Done

LonelyCpp commented on src/SCLoansEventEmitter.js:
```diff
@@ -0,0 +1,115 @@
+
+      this.subscriptions.push(subscription);
+      
+      console.log('SCLoansEvents Subscribed to gateway events');
```
remove all these console logs from the SDK

dporwal-shipit replied:
Done.

LonelyCpp commented on src/SCLoansEventEmitter.js:
```diff
@@ -0,0 +1,115 @@
+      try {
+        subscription.remove();
+        
+        const index = this.subscriptions.indexOf(subscription);
+        if (index > -1) {
+          this.subscriptions.splice(index, 1);
+        }
```
better to use `array.filter()`

dporwal-shipit replied:
Done.

LonelyCpp commented on src/SCGatewayEventEmitter.js:
```diff
@@ -0,0 +1,117 @@
+  USER_IDENTIFY: 'scgateway_user_identify',
+};
+
+export class SCGatewayEvents {
```
here again, a lot of the event emitter code is duplicated. 

structure it in a way where things can be shared as much as possible

dporwal-shipit replied:
We are currently treating both SDKs as separate and isolated. To support this approach, separate files have been created. The same pattern has also been followed in the Android implementation for the Notification Center.

dporwal-shipit replied:
Actually we are treating both the sdks as separate and isolated. So, created separate files to support this. Same is true in android implementation as well for Notification Center.

dporwal-shipit committed:
feature(analytics): making changes as requested in PR reviews

dporwal-shipit committed:
feature(analytics): making changes as per PR request reviews

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,388 @@
+                return true
+            }
+
+            // Add debug observer for all notifications
+            NotificationCenter.addObserver { notification ->
+                Log.d("DEBUG_ALL_EVENTS", "All notifications: ${notification.name}")
+            }
```
Remove

dporwal-shipit replied:
Removed

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,388 @@
+        const val USER_IDENTIFY = ScLoanNotification.USER_RESET
+    }
+
+    private var isListening = false
```
Can this just be derived?
```kotlin
private var isListening: Boolean
  get() = notificationObserver != null
```

dporwal-shipit replied:
Done

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,388 @@
+                
+                try {
+                    // Only process scloans_notification - single way to subscribe
+                    if (notification.name == "scloans_notification") {
```
Use constant from the native SDK.

dporwal-shipit replied:
ok

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,388 @@
+    fun getListeningStatus(promise: Promise) {
+        try {
+            val status = Arguments.createMap().apply {
+                putBoolean("isListening", isListening)
+                putBoolean("hasNotificationObserver", notificationObserver != null)
```
Can just keep `isListening`

dporwal-shipit replied:
kepping isListening and has made it a computed property on basis of notificationObserver != null

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,388 @@
+    }
+
+    @ReactMethod
+    fun emitTestEvent(eventType: String, testData: ReadableMap?, promise: Promise) {
+        try {
+            Log.d(TAG, "Emitting test event: $eventType")
+
+            val payload = Arguments.createMap().apply {
+                putString("type", eventType)
+                putBoolean("isTest", true)
+                testData?.let { putMap("data", it) }
+            }
+
+            sendEvent("scloans_notification", payload)
+            promise.resolve("Test event emitted successfully")
+
+        } catch (e: Exception) {
+            Log.e(TAG, "Error emitting test event", e)
+            promise.reject("TEST_EVENT_ERROR", e.message, e)
+        }
+    }
```
Not needed right?

dporwal-shipit replied:
Removed was for debugging.

isouravganguly commented on android/build.gradle:
```diff
@@ -149,8 +156,8 @@ def kotlin_version = getExtOrDefault('kotlinVersion')
-    implementation 'com.smallcase.gateway:sdk:4.2.2'
-    implementation 'com.smallcase.loans:sdk:3.1.1'
+    implementation 'com.smallcase.gateway:sdk-sdk-extract-mixpanel-from-sdk:4.2.2-3183-release'
+    implementation 'com.smallcase.loans:sdk-sdk-extract-mixpanel-from-sdk:3.1.1-72-release'
```
Please check the versions. This is probably local-device maven version numbers.

dporwal-shipit replied:
Yes we are testing internal builds right now, we would add the final version when it would get merged to main branch and a final android sdk would be released.

indrajit-roy-sc commented on ios/SCLoansEmitter.swift:
```diff
@@ -0,0 +1,228 @@
+        print("SCLoansEmitter: Skipping unsupported static event: \(name)")
+      }
+    }
+  }
```
Not being used

dporwal-shipit replied:
Removed was for debugging

dporwal-shipit committed:
feature(analytics): making changes as per PR request reviews

indrajit-roy-sc committed:
refactor(SCGatewayEvents): streamline initialization and subscription logic

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,406 @@
+
+    companion object {
+        const val TAG = "SCGatewayBridgeEmitter"
+        const val ANALYTICS_EVENT = ScgNotification.ANALYTICS_EVENT
+        const val SUPER_PROPERTIES_UPDATED = ScgNotification.SUPER_PROPS_UPDATED
+        const val USER_RESET = ScgNotification.USER_RESET
+        const val USER_IDENTIFY = ScgNotification.USER_IDENTIFY
```
remove

dporwal-shipit replied:
Making use of framework provided using `getConstants()` in android to export the constants

indrajit-roy-sc replied:
As we discussed. Lets' just use JS constants on the JS side and native SDK constants on the bridge.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,406 @@
+        const val USER_IDENTIFY = ScgNotification.USER_IDENTIFY
+    }
+
+    private var isListening = false
```
Not required. Can use as getter where `notificationObserver != null`

dporwal-shipit replied:
Done

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,406 @@
+    }
+
+    // Internal start listening method with debug logging
+    private fun startListeningInternal(): Boolean {
```
Remove method split. DO it in `startListening`.

dporwal-shipit replied:
Done. Now, using a single method startListening.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,406 @@
+
+    // Internal start listening method with debug logging
+    private fun startListeningInternal(): Boolean {
+        return try {
```
No try catch needed

dporwal-shipit replied:
Removed entirely and logic shifted to startListening()

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,406 @@
+                
+                try {
+                    // Only process scg_notification - single way to subscribe
+                    if (notification.name == SmallcaseGatewaySdk.SCG_NOTIFICATION_NAME) {
```
Not required

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,406 @@
+            }
+
+            // Add debug observer for all notifications
+            NotificationCenter.addObserver { notification ->
```
?

dporwal-shipit replied:
Removed. Was for debugging and testing whether events are coming or not in initial phases.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,406 @@
+                }
+            }
+
+            notificationObserver?.let { observer ->
```
`?.let` not required

indrajit-roy-sc replied:
If kotlin doesn't infer types here properly then keep it

dporwal-shipit replied:
Sure.

dporwal-shipit committed:
refactor(analytics): refactoring the native files in bridge to forward the stringified payload

dporwal-shipit committed:
refactor(analytics): refactoring the code, shifted App.tsx from class to functional components

dporwal-shipit committed:
refactor(analytics): working state of android scgateway & scloans

dporwal-shipit committed:
refactor(analytics): working state of ios scgateway & scloans

dporwal-shipit committed:
refactor(analytics): making use of native sdk exposed properties directly in React Native plugin

dporwal-shipit committed:
refactor(analytics): standardize class structure with formatting and proper ordering as per syntax

dporwal-shipit committed:
chore(analytics): Merge upgrade/react-native-0.79.4 into branch upgrade-react-native-0.79.4-dhruv-events:

- Resolved Podfile.lock merge conflicts by taking upgrade branch dependencies
- Fixed BOOL parameter syntax in RCT_REMAP_METHOD (removed incorrect pointer *)
- Updated package.json script as per incoming change

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,159 @@
+        const val ANALYTICS_EVENT = ScgNotification.ANALYTICS_EVENT
+        const val SUPER_PROPERTIES_UPDATED = ScgNotification.SUPER_PROPS_UPDATED
+        const val USER_RESET = ScgNotification.USER_RESET
+        const val USER_IDENTIFY = ScgNotification.USER_IDENTIFY
```
This is still not required ig

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,159 @@
+            "USER_RESET" to ScgNotification.USER_RESET,
+            "USER_IDENTIFY" to ScgNotification.USER_IDENTIFY
+        )
+    }
```
To be removed

dporwal-shipit replied:
Removed as per our discussion that we would not rely on native sdks for event names.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,156 @@
+        const val TAG = "SCLoansBridgeEmitter"
+
+        const val ANALYTICS_EVENT = ScLoanNotification.ANALYTICS_EVENT
+        const val SUPER_PROPERTIES_UPDATED = ScLoanNotification.SUPER_PROPS_UPDATED
```
To be removed

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,156 @@
+            "ANALYTICS_EVENT" to ScLoanNotification.ANALYTICS_EVENT,
+            "SUPER_PROPERTIES_UPDATED" to ScLoanNotification.SUPER_PROPS_UPDATED,
+        )
+    }
```
Remove

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,156 @@
+        } catch (e: Exception) {
+            Log.e(TAG, "Error in startListening", e)
+            promise?.reject("START_LISTENING_ERROR", e.message ?: "Unknown error", e)
+        }
```
So many `try-catch`s but none of the method calls actually throw. Don't simply use try catch for every method call. Remove all that are not required.

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,156 @@
+        } catch (e: Exception) {
+            Log.e(TAG, "Error scheduling stopListening on UI thread", e)
+            promise.reject("STOP_LISTENING_ERROR", e.message, e)
+        }
```
Again. Try catch.

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,156 @@
+        } catch (e: Exception) {
+            Log.e(TAG, "Error processing SCLoans notification", e)
+        }
+    }
```
Again. Try catch.

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,156 @@
+    }
+
+    private fun sendEvent(eventName: String, jsonString: String) {
+        try {
+            if (reactContext.hasActiveCatalystInstance()) {
+                reactContext
+                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+                    .emit(eventName, jsonString)
+            } else {
+                Log.w(TAG, "React context not active, cannot send event: $eventName")
+            }
+        } catch (e: Exception) {
+            Log.e(TAG, "Error sending event to React Native: $eventName", e)
+        }
+    }
```
tr catch

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on ios/SCGatewayEmitter.swift:
```diff
@@ -0,0 +1,134 @@
+@objc(SCGatewayEmitter)
+class SCGatewayEmitter: RCTEventEmitter {
+    
+    private struct Constants {
+        static let payloadKey = "payload"
+        static let stringifiedPayloadKey = "payload_str"
+        static let scgNotificationName = "scg_notification"
+        
+        static let analyticsEvent = "scgateway_analytics_event"
+        static let superPropertiesUpdated = "scgateway_super_properties_updated"
+        static let userReset = "scgateway_user_reset"
+        static let userIdentify = "scgateway_user_identify"
+    }
```
Why do we need this?

dporwal-shipit replied:
Removed the event constants.

indrajit-roy-sc commented on smart_investing_react_native/App.tsx:
```diff
@@ -1,69 +1,231 @@
+  [key: string]: any;
+}
+
+class App extends React.Component<AppProps, AppState> {
```
Reason for class component? Revert back to Functional component with `useEffect`.

dporwal-shipit replied:
Reverted.

indrajit-roy-sc commented on src/SCGatewayEventEmitter.js:
```diff
@@ -0,0 +1,133 @@
+
+const nativeModule = NativeModules.SCGatewayBridgeEmitter;
+
+export const SCGatewayEventTypes = {
+	ANALYTICS_EVENT: nativeModule?.ANALYTICS_EVENT || 'scgateway_analytics_event',
+	SUPER_PROPERTIES_UPDATED: nativeModule?.SUPER_PROPERTIES_UPDATED || 'scgateway_super_properties_updated',
+	USER_RESET: nativeModule?.USER_RESET || 'scgateway_user_reset',
+	USER_IDENTIFY: nativeModule?.USER_IDENTIFY || 'scgateway_user_identify',
+};
```
change to JS constants only as discussed

dporwal-shipit replied:
Done

indrajit-roy-sc commented on src/SCGatewayEventEmitter.js:
```diff
@@ -0,0 +1,133 @@
+			return null;
+		}
+
+		const subscription = this.eventEmitter.addListener(nativeModule?.SCG_NOTIFICATION || 'scg_notification', (jsonString) => {
```
Use JS constant

dporwal-shipit replied:
Done

indrajit-roy-sc commented on src/SCGatewayEventEmitter.js:
```diff
@@ -0,0 +1,133 @@
+
+				this.subscriptions = this.subscriptions.filter(sub => sub !== subscription);
+
+				if (this.subscriptions.length === 0) {
+					console.log('[SCGatewayEvents] No active subscriptions remaining, cleaning up');
+					this.cleanup();
+				}
```
This is not required here

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on src/SCLoansEventEmitter.js:
```diff
@@ -0,0 +1,129 @@
+
+const nativeModule = NativeModules.SCLoansBridgeEmitter;
+
+export const SCLoansEventTypes = {
+	ANALYTICS_EVENT: nativeModule?.ANALYTICS_EVENT || 'scloans_analytics_event',
+	SUPER_PROPERTIES_UPDATED: nativeModule?.SUPER_PROPERTIES_UPDATED || 'scloans_super_properties_updated',
+};
```
Remove native constants

dporwal-shipit replied:
Removed

indrajit-roy-sc commented on src/SCLoansEventEmitter.js:
```diff
@@ -0,0 +1,129 @@
+			return null;
+		}
+
+		const subscription = this.eventEmitter.addListener(nativeModule?.SCLOANS_NOTIFICATION || 'scloans_notification', (jsonString) => {
```
Remove native constant

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on src/SCLoansEventEmitter.js:
```diff
@@ -0,0 +1,129 @@
+
+				this.subscriptions = this.subscriptions.filter(sub => sub !== subscription);
+
+				if (this.subscriptions.length === 0) {
+					console.log('[SCLoansEvents] No active subscriptions remaining, cleaning up');
+					this.cleanup();
+				}
```
Not required

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on src/SCLoansEventEmitter.js:
```diff
@@ -0,0 +1,129 @@
+	 */
+	unsubscribeFromLoansEvent(subscription) {
+		if (subscription && typeof subscription.remove === 'function') {
+			try {
```
remove `try-catch`

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on src/SCLoansEventEmitter.js:
```diff
@@ -0,0 +1,129 @@
+	}
+
+	cleanup() {
+		try {
```
remove `try-catch`

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented:
Please remove all `try-catch` blocks where none of the internal function calls can throw.

dporwal-shipit committed:
refactor: improve event emitters and clean up codebase

- Fix Kotlin syntax errors and formatting issues in bridge emitters
- Remove unnecessary try-catch blocks for cleaner error handling
- Update App.tsx for functional components
- Replace SDK-supplied constants with JS constants for better consistency
- Improve code readability and maintainability

Affects:
- Android: SCGatewayBridgeEmitter.kt, SCLoansBridgeEmitter.kt
- iOS: SCGatewayEmitter.swift, SCLoansEmitter.swift
- JS: SCGatewayEventEmitter.js, SCLoansEventEmitter.js
- App: App.tsx

dporwal-shipit committed:
refactor: we no longer need try-catch in onCatalystInstanceDestroy

dporwal-shipit committed:
refactor: we no longer need try-catch in onCatalystInstanceDestroy

dporwal-shipit committed:
refactor: renaming to ensure consistency across ios/android

dporwal-shipit committed:
changed the PRODUCT_BUNDLE_IDENTIFIER and updated build.gradle with new android sdk

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,112 @@
+        
+        const val GATEWAY_NOTIFICATION = "scg_notification"
+        const val PAYLOAD_KEY = "payload"
+        const val STRINGIFIED_PAYLOAD_KEY = "payload_str"
```
These should be removed. Use the constants from the native sdk directly wherever required.

dporwal-shipit replied:
So, we would be keeping the event names on RN plugin hardcoded and these would still be inferred from native sdks. right?

indrajit-roy-sc replied:
Re Declaration of these constants  here is not required. Just use `SmallcaseGatewaySdk.SCG_NOTIFICATION_NAME` etc... directly in the code itself.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCGatewayBridgeEmitter.kt:
```diff
@@ -0,0 +1,112 @@
+        if (reactContext.hasActiveCatalystInstance()) {
+            reactContext
+                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+                .emit(eventName, jsonString)
```
This is one place we do need try catch.

dporwal-shipit replied:
here we are already checking through (reactContext.hasActiveCatalystInstance()).

Do we still need try-catch?

indrajit-roy-sc replied:
`.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)` has a chance of throwing. Eventhough slim, it's a good to have. Not a hard requirement.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,117 @@
+            
+            const val LOAN_NOTIFICATION = "scloans_notification"
+            const val PAYLOAD_KEY = "payload"
+            const val STRINGIFIED_PAYLOAD_KEY = "payload_str"
```
These should be removed. Use the constants from the native sdk directly wherever required.

dporwal-shipit replied:
So, we would be keeping the event names on RN plugin hardcoded and these would still be inferred from native sdks. right?

indrajit-roy-sc replied:
Re Declaration of these constants  here is not required. Just use `ScLoan.SCLOANS_NOTIFICATION_NAME` etc... directly in the code itself.

indrajit-roy-sc commented on android/src/main/java/com/reactnativesmallcasegateway/SCLoansBridgeEmitter.kt:
```diff
@@ -0,0 +1,117 @@
+            if (reactContext.hasActiveCatalystInstance()) {
+                reactContext
+                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+                    .emit(eventName, jsonString)
```
can put try catch here

dporwal-shipit replied:
here we are already checking through (reactContext.hasActiveCatalystInstance()).

Do we still need try-catch?

indrajit-roy-sc replied:
`.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)` has a chance of throwing. Eventhough slim, it's a good to have. Not a hard requirement.

indrajit-roy-sc commented on android/build.gradle:
```diff
@@ -57,6 +58,12 @@ android {
+  // Absolute or relative path to the app's RN Android directory
+  rootProject.ext.reactNativeAndroidRoot = file('../../smart_investing_react_native/node_modules/react-native/android')
+}
+
```
What is this? Remove.

dporwal-shipit replied:
Added while adding the mavenLocal generated sdk.

Removed.

indrajit-roy-sc commented on android/build.gradle:
```diff
@@ -8,6 +8,7 @@ buildscript {
   }
 
   dependencies {
+    //noinspection GradlePluginVersion
```
remove

dporwal-shipit replied:
Added while adding the mavenLocal generated sdk.

Removed.

indrajit-roy-sc commented on android/build.gradle:
```diff
@@ -69,7 +76,7 @@ repositories {
       }
   }
-  
+
```
redundant diff

indrajit-roy-sc commented on ios/SCGatewayEmitter.swift:
```diff
@@ -0,0 +1,123 @@
+@objc(SCGatewayEmitter)
+class SCGatewayEmitter: RCTEventEmitter {
+
+  private struct Constants {
+    static let payloadKey = "payload"
+    static let stringifiedPayloadKey = "payload_str"
+    static let scgNotificationName = "scg_notification"
+  }
```
These should be removed. Use the constants from the native sdk directly wherever required.

dporwal-shipit replied:
Done

indrajit-roy-sc replied:
Still there

dporwal-shipit replied:
I would remove these after making a patch release. right?

indrajit-roy-sc replied:
yes

dporwal-shipit replied:
Done

indrajit-roy-sc commented on ios/SCLoansEmitter.swift:
```diff
@@ -0,0 +1,123 @@
+@objc(SCLoansEmitter)
+class SCLoansEmitter: RCTEventEmitter {
+
+  private struct Constants {
+    static let loanNotification = "scloans_notification"
+    static let payloadKey = "payload"
+    static let stringifiedPayloadKey = "payload_str"
+  }
```
These should be removed. Use the constants from the native sdk directly wherever required.

dporwal-shipit replied:
Right now, not possible. Would try to accommodate in next release

indrajit-roy-sc replied:
The code is already merged on ios. Make a patch release.

dporwal-shipit replied:
I would remove these after making a patch release. right?

dporwal-shipit replied:
Done

indrajit-roy-sc commented on smart_investing_react_native/metro.config.js:
```diff
@@ -1,11 +1,21 @@
+      'react-native-smallcase-gateway': path.resolve(__dirname, '../'),
+    },
+  },
+};
```
required?

dporwal-shipit replied:
Removed.

indrajit-roy-sc commented on types/index.d.ts:
```diff
@@ -46,6 +50,13 @@ declare const _default: {
+    subscribeToGatewayEvents: (callback: (event: any) => void) => any;
+    unsubscribeFromGatewayEvents: (subscription: any) => void;
+    subscribeToLoansEvent: (callback: (event: any) => void) => any;
+    unsubscribeFromLoansEvent: (subscription: any) => void;
```
These types need to be generated properly. Take @LonelyCpp 's help.

dporwal-shipit replied:
Asked and Updated.

indrajit-roy-sc commented on types/index.d.ts:
```diff
@@ -1,4 +1,8 @@
+export { ScLoan, SCGatewayEventManager, SCLoansEventManager };
+
+declare const SCGatewayEventManager: any;
+declare const SCLoansEventManager: any;
```
These types need to be generated properly. Take @LonelyCpp 's help.

dporwal-shipit replied:
Asked and Updated.

dporwal-shipit committed:
making changes as per PR reviews

indrajit-roy-sc commented on smart_investing_react_native/android/build.gradle:
```diff
@@ -32,7 +32,22 @@ allprojects {
-        maven { url "${artifactory_contextUrl}/SCGateway" }
+        maven { 
+            url "${artifactory_contextUrl}/SCGateway"
+            credentials {
+                username "${artifactory_user}"
+                password "${artifactory_password}"
+            }
```
Not required for `SCGateway`

dporwal-shipit committed:
making changes as per PR

dporwal-shipit committed:
Merge remote-tracking branch 'origin/development' into upgrade-react-native-0.79.4-dhruv-events

indrajit-roy-sc commented on ios/SCGatewayEmitter~.swift:
Duplicate file

indrajit-roy-sc commented on ios/SCLoansEmitter.swift:
```diff
@@ -0,0 +1,118 @@
+
+      self.stopListening()
+
+      let notificationName = Notification.Name(ScLoan.scLoansNotificationName.rawValue)
+
+      self.notificationObserver = NotificationCenter.default.addObserver(
+        forName: notificationName,
```
```suggestion
      self.notificationObserver = NotificationCenter.default.addObserver(
        forName: ScLoan.scLoansNotificationName,
```

dporwal-shipit replied:
Done

dporwal-shipit committed:
2nd emitter got added while testing in XCode-VSCode, removed

dporwal-shipit committed:
making changes as per PR reviews

dporwal-shipit committed:
resolving conflicts while merging react-native-events branch to development

dporwal-shipit committed:
resolving conflicts while merging react-native-events branch to development
